core/src/Revolution/Processors/System/DownloadOutput.php

```diff
@@ -23,69 +24,44 @@ class DownloadOutput extends Processor
      */
     public function process()
     {
-        if ($this->getProperty('download')) {
-            $output = $this->download();
-        } else {
-            $output = $this->cache();
+        $cookieName = $this->getProperty('cookieName');
+        if ($cookieName) {
+            setcookie($cookieName, 'true', time() + 10, '/');
         }
-        return $output;
-    }
-
-    /**
-     * Download the output to the browser
-     * @return string
-     */
-    public function download()
-    {
-        $dl = $this->getProperty('download');
-        $dl = str_replace(['../', '..', 'config'], '', $dl);
-        $dl = ltrim($dl, '/');
-
-        $f = $this->modx->getOption('core_path') . $dl;
-        $o = $this->modx->cacheManager->get($dl);
-        if (!$o) {
-            return '';
-        }
-
-        $this->modx->cacheManager->delete($dl);
 
-        $bn = basename($f);
-        @session_write_close();
-        header('Content-Type: application/force-download');
-        header('Content-Disposition: attachment; filename="' . $bn . '-' . date('Y-m-d Hi') . '.txt"');
-        return $o;
+        return $this->download();
     }
 
     /**
-     * Cache the data stored
-     * @return array|string
+     * Create a temporary file object and serve as a download to the browser
+     *
+     * @return bool|string
      */
-    public function cache()
+    public function download()
     {
         $data = $this->getProperty('data');
+
         if (empty($data)) {
             return $this->failure($this->modx->lexicon('invalid_data'));
         }
 
         $data = strip_tags($data, '<br><span><hr><li>');
         $data = str_replace(['<li>', '<hr>', '<br>', '<span>', '<?php', '<?', '?>'], "\r\n", $data);
         $data = strip_tags($data);
-        $o = '/*
-* MODX Console Output
-*
-* @date ' . date('Y-m-d H:i:s') . '
-*/
-' . $data . '
-/* EOF */
-';
+        $output = "/*\r\n* MODX Console Output\r\n*\r\n* @date " . date('Y-m-d H:i:s') . "\r\n*/\r\n" . $data . "\r\n/* EOF */\r\n";
```

Comment:
Maybe we should use the system setting to display the time format? Or is it not needed here?

Reply:
The system settings are based on javascript date format. This is code is PHP.